### PR TITLE
[GROW-141] Destination MySqlInser

### PR DESCRIPTION
**Pull request by dianadevargas**

### CODE REVIEW

File: src/destinations/MySqlInsert.ts [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        }
+    }
+
+    public async processRecords(batch: EtlBatch): Promise<boolean> {
+        const fieldList = batch.getTransformedRecords().map( (record) => record.getTransformedData() );
+        const deleteAllRecords = this.deleteAllRecordsBeforeLoad && (batch.getBatchNumber() === 1);
+        return await this.mySqlClient.runInTransaction(false, async (transaction: DBTransaction) => {
+            if (deleteAllRecords) {
+                log.debug(`Delete all records from table : ${this.tableName}`);
+                await transaction.query(`delete from ${this.tableName}`);
```
Comment (chungryan) [inline src/destinations/MySqlInsert.ts]: whilst this works, truncating the table is much faster.
so depending on how many records you are deleting, may be worth doing that instead

File: src/destinations/MySqlInsert.ts [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    public getQueryToDeleteRows(inputs: Array<object>, deleteFields: Array<string>): MysqlQueryBuilderHelper {
+        const deleteQuery = {
+            queryValues: [],
+            query: '',
+        };
+        if (deleteFields.length === 1) {
+            const field = deleteFields.pop();
+            deleteQuery.queryValues = this.getFieldListValues(inputs, field);
+            const maskText = deleteQuery.queryValues.map((deleteValue) => {
+                return '?';
```
Comment (chungryan) [inline src/destinations/MySqlInsert.ts]: you can use the shortcut instead: map(() => '?')

File: src/index.ts [hunk 3]
```diff
@@ -24,3 +24,5 @@ export { GoogleAnalytics } from './util/GoogleAnalytics';
 export { mapRecord } from './util/FieldMapper';
 export { MySQLDataByKey } from './sources/MySQLDataByKey';
 export { StaticSavepointManager } from './savepoints/StaticSavepointManager';
+export { MySqlInsert } from './destinations/MySqlInsert';
+export { GoogleAnalyticsReporting } from './sources/GoogleAnalyticsReporting';
```
Comment (chungryan) [inline src/index.ts]: good one!

File: src/destinations/MySqlInsert.ts (hunk 1 above)
Comment (dianadevargas) [reply src/destinations/MySqlInsert.ts]: True, but we can't truncate in most of the databases because only the root user has permission to truncate tables

**Pull request merged**